### DISCUSSION TITLE
idl message convert to dynamic data in 2.6.9

### DISCUSSION THREAD

**Discussion by com-server** [eProsima/Fast-DDS#5344] [Q&A]
Dose anyone know how to convert the idl message to dynamic data?  such as , convert std_msgs::msg::String msg to the dynamic data.
if so ,please give a example. thanks a lot .

**Comment by juanlofer-eprosima** [comment]
Hi @com-server ,

Two serialization helpers were introduced in Fast-DDS 3.0, a DynamicType to IDL converter, and a DynamicData to JSON converter. Please have a look at [Serialization Utilities](https://fast-dds.docs.eprosima.com/en/latest/fastdds/xtypes/type_serializing.html) for more details. 

But I believe your are asking about converting a JSON object (e.g. obtained with `json_serialize`) to DynamicData, feature on which we are working at the moment.

**Reply by com-server-ap** [reply]
Two serialization helpers were introduced in Fast-DDS 3.0, a DynamicType to IDL converter, and a DynamicData to JSON converter. Please have a look at [Serialization Utilities](https://fast-dds.docs.eprosima.com/en/latest/fastdds/xtypes/type_serializing.html) for more details.
    --- how about the version 2.6.9?
 But I believe your are asking about converting a JSON object (e.g. obtained with json_serialize) to DynamicData, feature on which we are working at the moment.
    yes, i just want to convert static mesage to json object and json object to static message.  but i find that we must convert to DynamicData first.  have you already implemented this? could you share your thought process ? Thanks again.

**Reply by juanlofer-eprosima** [reply]
Regarding your first question, no these utilities are and will not be included in any 2.x version as its implementation depends on the deep XTypes refactor introduced in 3.0 release.

Answering your second question, it is not in the road map to implement a function capable of converting a static message to JSON. However, once the JSON to DynamicData converter function is available, it should be possible to achieve so although in a rather involved mannner (static data -> serialized payload (static PubSubTypes serialize) -> DynamicData (dynamic PubSubTypes deserialize) -> JSON (new method)).

**Reply by jfergusonmi** [reply]
What is the status of this? I am using json_serialize in 3.1.0 to encode DynamicData to a json string. Is there still not an API to convert json string to DynamicData? If not, is there any public classes that are available we could use to do this or would it have to be manually converted ourselves?

**Reply by juanlofer-eprosima** [reply]
We are working on the feature. It is expected to be available by the end of the year but since we have no deadline for this it may be delayed some more time to cover other priorities.

**Reply by juanlofer-eprosima** [reply]
@com-server @jfergusonmi The requested feature is now available in [Fast DDS v3.3.0](https://github.com/eProsima/Fast-DDS/releases/tag/v3.3.0). Please have a look at the corresponding [documentation section](https://fast-dds.docs.eprosima.com/en/v3.3.0/fastdds/xtypes/type_serializing.html#json-to-dynamicdata) for more information.

**Reply by yaseiva** [reply]
The release is deleted?

**Reply by juanlofer-eprosima** [reply]
Hi @yaseiva, yes we temporally removed the tag to review a potential issue, but it should be out again very soon. You can still access the branch [here](https://github.com/eProsima/Fast-DDS/tree/3.3.0).

**Reply by yaseiva** [reply]
Today? is it possible to get my hands on the pre-release installation file? instead of source code. Thanks

**Reply by juanlofer-eprosima** [reply]
We hope it will be ready for today, but if not it should be out early next week. And no, unfortunately we cannot distribute the installation files yet.
